NitroxClient/GameLogic/Constructor.cs

```diff
@@ -35,8 +36,40 @@ public void BeginCrafting(GameObject constructor, TechType techType, float durat
                 GameObject constructedObject = (GameObject)opConstructedObject.Get();
 
                 List<InteractiveChildObjectIdentifier> childIdentifiers = VehicleChildObjectIdentifierHelper.ExtractGuidsOfInteractiveChildren(constructedObject);
+                Vehicle vehicle = constructedObject.GetComponent<Vehicle>();
                 string constructedObjectGuid = GuidHelper.GetGuid(constructedObject);
-                ConstructorBeginCrafting beginCrafting = new ConstructorBeginCrafting(constructorGuid, constructedObjectGuid, techType, duration, childIdentifiers, constructedObject.transform.position, constructedObject.transform.rotation);
+               
+                //Initialize some default values to serialize.
+                Vector3[] HSB = new Vector3[5];
+                Vector3[] Colours = new Vector3[5];
+                Vector4 tmpColour = Color.white;
+                string name = "Cyclops"; // Cant find a way to actually get the Cyclops name.
+
+                if (!vehicle)
```

Comment:
maybe break this into a new function such as `getVehicleAttributes(vehicle, out hsb, out colours, out name)`